_posts/2020-12-22-aws-antipatterns.md

```diff
@@ -0,0 +1,198 @@
+
+!['Allow everything' - policy](/img/aws-antipatterns/crazy_wide_policy.png)
+
+The wider concept here is to minimize the **blast radius**. And we do that with the Principle of Least Permission (**POLP**). Blast radius means an area of services and infrastructure that may be affected if a security breach or misconfiguration takes place. In other words, if your lambda has full access to anything in the account, and it gets compromised, damage can be horrendous. Also, if anyone is auditing your solution, this mess will be revealed and require actions to fix.
+
+![Allow some things for ALL s3 buckets](/img/aws-antipatterns/wildcard_policy.png)
+
+Symptoms of too wide a blast radius are:
+- AWS Policies with a lot of asterisks giving wide or unlimited access to services, or resources
+- AWS Roles using built-in system policies with FullAccess permissions
+- Developers doing their daily work with credentials that use similarly unlimited policies
+- No network isolation, everything in one network
+
+So, how do you fix this? Well, you invest some time, effort, and perspiration. You create those boundaries and walls. You create minimal permissions that are required to do the task. If you create them in a good manner, they do not limit the valid uses of software, but they limit the damage if an 'explosion' happens within. They limit the width of a potential exploit. They also document what parts of services you are using. These boundaries are much cheaper and easier to create while building the solution, and they can be costly to fix afterward. So the second your POC grows to an MVP, and you start doing implementation sprints, you should start doing the POLP dance, including it in the code and review workflow.
+
+There are some tools also to help with this, that can review the templates or policies, and alert if they seem a bit too wide. For example, AWS has services like IAM access advisor, or AWS Trusted Advisor/AWS Organizations can be used to audit this constantly yourself. I've also had success with tools such as CFRipper. There are also plenty of commercial tools if you wish to go that route. You can run these as part of your build, or set up a Cloudformation event-triggered lambda that can stop deployment if the quality gate is not passed. With good tools, it's possible to create exceptions to rules or modify and evolve the rules yourself. But a good tool is also cost-effective and cheap. If you introduce some automation in your development process, this cleanup and tightening are automatically done when it's the cheapest.
```

Comment:
Maybe add link to this?
https://aws.amazon.com/blogs/security/tighten-s3-permissions-iam-users-and-roles-using-access-history-s3-actions/

Reply:
Perhaps not, this is steering in the domain of ops work, I try to keep on general advice level, again we could go deeper and more concrete on any topic but here I would choose not to.